Implementation/Services/index.md

```diff
@@ -0,0 +1,510 @@
+
+:::warning
+Although there is a Management Service compelling called the `ContentService` - only use this to modify content - do not use the `ContentService` in a View/Template to pull back data to display, this will make database requests and be slow - here instead use the generically named `UmbracoHelper` to access the `PublishedContentQuery` methods that operate against a cache of published content items, and are significantly quicker.
+:::
+
+The Management Services and Helpers are all registered with Umbraco's underlying DI framework - this article aims to show examples of gaining access to utilise these resources in multiple different scenarios (there are subtle differences to be aware of depending on what part of Umbraco is being extended) - and also suggest how to follow a similar pattern to encapsulate custom 'site specific' implementation logic, in similar services and helpers, registered with the underlying DI contain, to avoid repetition and promote consistency and readability within an Umbraco site solution.
+
+## Accessing Management Services and Helpers in a Template/View
+
+Inside a view/template or partial view that inherits from UmbracoViewPage, access is provided to Umbraco's services via the `Services` [service context](../../Reference/Management/Services/) and the `Umbraco` - [UmbracoHelper](../..//Reference/Querying/UmbracoHelper) - is a gateway to an array of useful functionlity.
+
+```csharp
+@inherits UmbracoViewPage<Blogpost>
+@using ContentModels = Umbraco.Web.PublishedModels;
+@{
+    Layout = "master.cshtml";
+
+    // retrieve an item from Umbraco's published cache with id 123
+    IPublishedContent publishedContentItem = Umbraco.Content(123);
+
+    // it is 'unlikely' to need to use a Management Service in a view:
+    var relationService = Services.RelationService;
+}
+```
+## Accessing Management Services and Helpers in a Controller
+
+Inside a [custom controller](../../Reference/Routing/custom-controllers) you can inject any of the core Management Services or Helpers required to assist in controlling the custom flow of the particular request via the Controller's constructor:
+
+```csharp
+using System.Collections.Generic;
+using System.Web.Mvc;
+using Umbraco.Web.Models;
+using Umbraco8.Services;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Services;
+
+namespace Umbraco8.Controllers
+{
+    public class BlogPostController : Umbraco.Web.Mvc.RenderMvcController
+    {        // GET: BlogPost   
+        private readonly ILogger _logger;
+        private readonly UmbracoHelper _umbracoHelper;
+        private readonly RelationService _relationService;
+
+        public BlogPostController(ILogger logger, UmbracoHelper umbracoHelper, RelationService relationService)
```

Comment:
Yes, I've reworked the article in several places to reflect this - we need to document for those that choose to use DI for the full DI experience leading through to unit testability and beyond, and also for those who perhaps choose not to use it to that extent, but will gain benefit from the simplicity of the Service Locator fallback 'just working' for missing constructors, and hopefully can still be encouraged to utilise grouping common implementation code into a service...

... I've tried to think about how best to try and convey this to different audiences...

what I have is two examples, the first showing the fallback 'just works', followed by a warning regarding DI followed by the proper best practice for DI approach - and added the tip and the gif to generate them!

Hopefully this doesn't scare people off with a ton of constructors they've never seen before, but lures them into the 'why' and where 'it could go' with DI - but easy to delete the first bit if it's felt the documentation shouldn't try to engage in this way.

I guess in some implementations the length of the constructors could get quite long... so yes +1 for, is it a Facade Service?, or SuperControllerContextObject